### DISCUSSION TITLE
Autoplay Videos on IOS

### DISCUSSION THREAD

**Discussion by seanaye** [ionic-team/capacitor#3518] [Q&A]
How can I enable autoplaying of videos on IOS? when I try to autoplay a video I receive 

```
Unhandled Promise Rejection: NotAllowedError: The request is not allowed by the user agent or the platform in the current context, possibly because the user denied permission.
```
in the console.

This is not an issue on android

**Comment by dotNetkow** [comment]
Code and steps you've tried?

**Reply by seanaye** [reply]
Whoops, forgot that part sorry.

I'm using intersection observer to play videos when they are scrolled into view

```typescript
    const intersectionOptions = {
      handler: (entry: IntersectionObserverEntry) => {
        if (entry.intersectionRatio > 0.9) {
          context.emit('loadNext')
        }
        if (entry.isIntersecting) {
          context.emit('scroll', props.index)
          play()
        } else {
          pause()
        }
      },
      cfg: {
        threshold: [0.5, 1.0]
      }
    }
```
this `play()` function is just a wrapper around native html5 `videoElem.play()`

This works like a charm on capacitor android and chrome desktop, but throws the above error on capacitor ios.

Not sure how to fix this as it appears to be subject to safari's video autoplay rules although this should be bypassed in capacitor according to #1094

**Reply by dotNetkow** [reply]
Makes sense, thanks. Given #1094, you may need to open a new issue, providing as much of a repro as possible (code, etc.).

**Comment by roxteddy** [comment]
Hi !

#1094 gives the ability to play video without UI interaction but wether you use attribute autoplay or try to start video programmatically you might need to :

- use **playsinline** attribute on video (allows the video to play inside web element - not fullscreen)
- use **muted** attribute (I don't think you can autoplay with sound on)

Does it help ?

**Comment by vikicoder007** [comment]
Nope chrome and other browsers i think block videos autopay don't know how youtube managed to autoplay with sound
